XcodeServerSDK/Server Entities/Device.swift

```diff
@@ -10,21 +10,55 @@ import Foundation
 
 public class Device : XcodeServerEntity {
     
-    public let name: String
-    public let simulator: Bool
     public let osVersion: String
-    public let deviceType: String
     public let connected: Bool
+    public let simulator: Bool
+    public let modelCode: String // Enum?
+    public let deviceType: String // Enum?
+    public let modelName: String
+    public let deviceECID: String?
+    public let modelUTI: String
+    public let activeProxiedDevice: Device?
+    public let docType: String
+    public let trusted: Bool
+    public let name: String
+    public let supported: Bool
+    public let processor: String?
+    public let identifier: String
+    public let enabledForDevelopment: Bool
+    public let serialNumber: String?
     public let platform: DevicePlatform.PlatformType
+    public let architecture: String // Enum?
+    public let isServer: Bool
+    public let retina: Bool
     
     public required init(json: NSDictionary) {
         
-        self.name = json.stringForKey("name")
-        self.simulator = json.boolForKey("simulator")
         self.osVersion = json.stringForKey("osVersion")
-        self.deviceType = json.stringForKey("deviceType")
         self.connected = json.boolForKey("connected")
+        self.simulator = json.boolForKey("simulator")
+        self.modelCode = json.stringForKey("modelCode")
+        self.deviceType = json.stringForKey("deviceType")
+        self.modelName = json.stringForKey("modelName")
+        self.deviceECID = json.optionalStringForKey("deviceECID")
+        self.modelUTI = json.stringForKey("modelUTI")
+        if let proxyDevice: NSDictionary = json.optionalForKey("activeProxiedDevice") {
```

Comment:
You can just use `if let proxyDevice = json.optionalDictionaryForKey("activeProxiedDevice") {`